gputop/remote/gputop-ui.js

```diff
@@ -191,14 +165,19 @@ GputopUI.prototype.update_graphs = function(timestamp) {
             var end = counter.updates[j][1];
             var val = counter.updates[j][2];
             var max = counter.updates[j][3];
-
             var mid = start + (end - start) / 2;
 
-            /* FIXME we should stop assuming all counters are a percentage */
-            var percent = 100 * val / max;
-
-            counter.graph_data.push([mid, percent]);
-
+            if (counter.duration_dependent && counter.latest_duration) {
+                var per_sec_scale = 1000000000 / counter.latest_duration;
+                val *= per_sec_scale;
+                max *= per_sec_scale;
+                if (max > counter.inferred_max)
+                    counter.inferred_max = max;
+                if (val > counter.inferred_max)
+                    counter.inferred_max = val;
```

Comment:
We shouldn't be updating counter.inferred_max here and in gputop.js:append_counter_data()

We should update append_counter_data() to convert d_value and max into per-second units (if duration dependent) before updating counter.latest_value/latest_max/latest_duration/inferred_max.

gputop-ui.js:update_counter() should then also be updated to not repeat the conversion to per-second units.